### DISCUSSION TITLE
**Help** Configure hyperlink in email alert body

### DISCUSSION THREAD

**Discussion by ngms17** [jertel/elastalert2#1220] [Q&A] [answered]
Hi,

Currently i have elastalert2 configured to run an aggregation of certain alerts and then send an email.

I am using a custom Kibana discover page that i generate a link and the pass it on "alert_text" inside the rule configuration file. This link is dynamic because it is using the value of the "suricata_flow_id" of each alert of the aggregation. But this link is to big and it is being shown in the email body as:

`LINK: https://kibana:5601/app/discover#/view/13dd22f0-86cc-11e8-b59d-21efb914e65c-ecs?_g=(filters:!(),refreshInterval:(pause:!t,value:10000),time:(from:'2023-07-19T13:42:10.399000',to:now))&_a=(columns:!(suricata.eve.flow_id,suricata.eve.event_type,suricata.eve.alert.signature,source.ip,source.port,destination.ip,destination.port),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'suricata-ids-*',key:event.kind,negate:!f,params:!(event,alert),type:phrases,value:!(event,alert)),query:(bool:(minimum_should_match:1,should:!((match_phrase:(event.kind:event)),(match_phrase:(event.kind:alert)))))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'suricata-ids-*',key:event.module,negate:!f,params:(query:suricata),type:phrase),query:(match_phrase:(event.module:(query:suricata)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'suricata-ids-*',key:suricata.eve.flow_id,negate:!f,params:(query:'1728895498860685'),type:phrase),query:(match_phrase:(suricata.eve.flow_id:'1728895498860685')))),grid:(),hideChart:!f,index:'suricata-ids-*',interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))`t

Is it possible to configure this link to be shown as a word? Like this : 

LINK: [Some text](url)

I have tried this with "a hef" but with no success.

Thanks

**Comment by jertel** [comment]
Does the [Discover URL generator](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#generate-kibana-discover-url) built into ElastAlert 2 not work for you? If it did you could use the [Kibana URL shortener](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#shorten-kibana-discover-url) feature built into ElastAlert 2. 

Regarding your attempt to hide the link, are you specifying that the email format should be in HTML? Ex: `email_format: html` ([documented here](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#email)).

**Reply by ngms17** [reply]
I cant get Discover URL generator to work because of my necessity to use a dynamic link that fetches a value for a field (suricata.eve.flow_id).

I will try to specify the `email_format` to html and see if it works

**Reply by ngms17** [reply]
Hiding the link works with `email_format` as html. Now i have to construct my own html inside `alert_text`, right?

Although the aggregation table is unformatted. How can i solve the table issue?

**Reply by ngms17** [reply]
I was able to get it to work with HTML. Although, my aggregation results are appearing in two different tables. Is it possible to use "summary_table" in HTML?

Or maybe some way to create a HTML table that can display the aggregation results together as in the "summary_table" in ascii?

Another question, what field is used in "summary_table" to display the "count" information?

**Comment by jertel** [comment, marked as answer]
> I was able to get it to work with HTML. Although, my aggregation results are appearing in two different tables. Is it possible to use "summary_table" in HTML?

I've not tried. Most likely it wouldn't format correctly without using `<PRE>...</PRE>` tags wrapped around it. It may be possible to do this using a Jinja2 template, but again I haven't tried it.

> Or maybe some way to create a HTML table that can display the aggregation results together as in the "summary_table" in ascii?

Using the PREformatted tags as mentioned above would allow you to continue using ascii mode. Assuming it's possible to get those PRE tags in there.
 
> Another question, what field is used in "summary_table" to display the "count" information?

Aggregation queries return the counts for each bucket. It's stored in a dict object, where the key is a list of keys making up the compound aggregation (or just one key if a simple aggregation is requested), and the value is the count for that aggregation bucket. Code is located here: https://github.com/jertel/elastalert2/blob/33d0d1f0730ecdb5538b591ca800ccc67f4d078f/elastalert/alerts.py#L283

**Reply by ngms17** [reply]
I think using the PRE tags does not work. Maybe it should be interesting in a future version the possibilty of use summary_table as html.

Thanks!

**Reply by jertel** [reply]
Ok, thanks for the confirmation.
